arch/arm/src/armv7-a/cp15_clean_dcache_all.S

```diff
@@ -91,12 +91,16 @@
 
 cp15_flush_dcache_all:
 
+	push		{r4}
+
 	mrc		CP15_CCSIDR(r0)			/* Read the Cache Size Identification Register */
+	mov		r4, r0
+
 	ldr		r3, =0x7fff			/* Isolate the NumSets field (bits 13-27) */
 	and		r0, r3, r0, lsr #13		/* r0=NumSets (number of sets - 1) */
 
 	ldr		r3, =0x3ff			/* Isolate the way field (bits 3-12) */
-	add		r4, r3, r0, lsr #3		/* r4=(number of ways - 1) */
+	add		r4, r3, r4, lsr #3		/* r4=(number of ways - 1) */
```

Comment:
let's reuse r3 to avoid push/pop r4?

Reply:
@zhuyanlinzyl @xiaoxiang781216 
I agree to remove r4 and reuse some register (perhaps r1?).
By the way, I think ```add r4, r3, r0, lsr #3``` means ```r4 = r3 + (r0 >> 3)```.
I thought it would be ```and r4, r3, r0, lsr #3  (i.e. r4 = r3 & (r0 >> 3)```

Reply:
@masayuki2009 could you try the update?

Reply:
I see that comment about `and` vs `add` is not answered. I would appreciate if author can write here the answer